src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,48 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<K, V> {
+    private K first;
+    private V second;
+
+    private Pair(K first, V second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <K, V> Pair<K, V> of(K first, V second) {
+        return new Pair(first, second);
+    }
+
+    public K getFirst() {
+        return this.first;
+    }
+
+    public V getSecond() {
+        return this.second;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || this.getClass() != obj.getClass()) {
+            return false;
+        }
+        Pair<K, V> pair = (Pair<K, V>) obj;
+        return (this.first == null && pair.first == null ? true :
+                this.first != null ? this.first.equals(pair.first) : false)
+                &&
+                (this.second == null && pair.second == null ? true :
+                        this.second != null ? this.second.equals(pair.second) : false);
```

Comment:
this looks rather complicated
let's look at our previous task and do something similar that we did for rectangle

Reply:
done